drivers/flash/flash_stm32_qspi.c

```diff
@@ -100,6 +125,53 @@ static inline void qspi_unlock_thread(const struct device *dev)
 	k_sem_give(&dev_data->sem);
 }
 
+static inline void qspi_set_address_size(const struct device *dev,
+					 QSPI_CommandTypeDef *cmd)
+{
+	struct flash_stm32_qspi_data *dev_data = DEV_DATA(dev);
+
+	if (dev_data->flag_access_32bit) {
+		cmd->AddressSize = QSPI_ADDRESS_32_BITS;
+		return;
+	}
+
+	cmd->AddressSize = QSPI_ADDRESS_24_BITS;
+}
+
+static inline void qspi_prepare_quad_read(const struct device *dev,
+					  QSPI_CommandTypeDef *cmd)
+{
+	struct flash_stm32_qspi_data *dev_data = DEV_DATA(dev);
+
+	if (IS_ENABLED(STM32_QSPI_QUAD_IO) && dev_data->flag_quad_io_en) {
+		cmd->Instruction = dev_data->qspi_read_cmd;
+		cmd->AddressMode = QSPI_ADDRESS_4_LINES;
+		cmd->DataMode = QSPI_DATA_4_LINES;
+		cmd->DummyCycles = dev_data->qspi_read_cmd_latency;
+	}
+}
+
+static inline void qspi_prepare_quad_program(const struct device *dev,
+					     QSPI_CommandTypeDef *cmd)
+{
+	struct flash_stm32_qspi_data *dev_data = DEV_DATA(dev);
+	/*
+	 * There is no info about PP/4PP command in the SFDP tables,
+	 * hence it has been assumed that NOR flash memory supporting
+	 * 1-4-4 mode also would support fast page programming.
+	 */
+	if (IS_ENABLED(STM32_QSPI_QUAD_IO) && dev_data->flag_quad_io_en) {
```

Comment:
I think that the `IS_ENABLED` macro, although will work here, is intended to be used with `CONFIG_` definitions.

Reply:
I would prefer to leave it here.

Reply:
The checkpatch is shy to inform you that it would not like that: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/checkpatch.pl#L6511-L6515

Reply:
Ok, Then I will use:
`if (STM32_QSPI_QUAD_IO && dev_data->flag_quad_io_en)` directly in the if clause.